tests/slack_notifier/test_slack_notifier.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import json
+import io
+from collections import OrderedDict
+
+from lux.utils import test
+
+__test__ = True
+
+
+class TestSlackNotifier(test.TestCase):
+    
+    
+    def setUp(self):
+        from .utils import SlackTestHandler
+        self.teststream = io.StringIO()
+        # replacing app/server loggers
+        self.logger = logging.getLogger("TestLogger")
+        self.logger.lvl = 30
+        self.handler = SlackTestHandler(self.teststream)
+        self.handler.level = 30
+        self.handler.setFormatter(logging.Formatter(json.dumps(
+                                OrderedDict([
+                                ("text", '%(name)s - %(message)s'),
+                                ("channel", "#testchannel"),
+                                ("username", "testname"),
+                                ("icon_emoji", ":testicon:")]))))
+        self.logger.addHandler(self.handler)
+    
+    def read(self, stream):
+        # the StringIO stream adds a \n at the end of every line and
+        # adds \00 at the beginning of some of them; strip is used to 
+        # remove these artifacts coming from the chosen stream
+        a = stream.getvalue().strip("\n").strip("\00")
+        stream.truncate(0)
+        stream.flush()
+        return a
+    
+    def test_formatting(self):
+        testrecord = logging.makeLogRecord({"name": "TestLogger",
```

Comment:
why you use `makeLogRecord` instead of `log` or better `info`, `warning`, `error` etc.?